File: C++Code/1051.cpp
//Questão: Imposto de renda
#include <iostream>
#include <iomanip>

using namespace std;

int main(){
    double a,taxa,taxa1,taxa2,total;
    cin >> a;
    
    if (a >= 0 && a <= 2000.00)
    {
       cout << "Isento\n";
    }
    else
    {
        if (a > 2000 && a <= 3000)
        {
            total = ((a - 2000)*8)/100;
        }
        else if (a > 3000 && a <= 4500)
        {
            taxa = a - 3000;
            taxa1 = (1000 *8)/100;
            total = ((taxa * 18)/100) + taxa1;
        }
        else if (a > 4500)
        {
            taxa = (a - 4500);

            taxa1 = (1000 *8)/100;
            taxa2 = (1500 * 18)/100;
            total = ((taxa * 28)/100) +taxa2+taxa1;

        }
        cout<<fixed<<setprecision(2)<< "R$ "<<total<< endl;
    }
    
    
    
    return 0;
}

File: PythonCode/1051.py
#Questão: Imposto de renda
a = float(input())
if (a > 0.00) and (a <= 2000.00):
    print('Isento')
elif (a > 2000.00) and (a <= 3000.00):
    taxa = (a - 2000.00)
    total = ((taxa * 8)/100)
    print('R$ {:.2f}'.format(total))
elif (a > 3000.00) and (a <= 4500.00):
    taxa = (a - 2000.00)
    total1 = (taxa -1000.00)
    taxa2 = ((1000.00 * 8)/100)
    taxa3 = ((total1 * 18)/100)
    total = taxa3 + taxa2
    print('R$ {:.2f}'.format(total))
elif (a > 4500.00):
    #taxa = (a - 2000.00)
    #total1 = (taxa - 1000.00)

    taxa2 = ((1000.00 * 8)/100)
    #total2 = (total1 - 1500.00)
    
    taxa3 = ((1500.00 * 18)/100)
    taxa4 = ((total2 * 28)/100)

    total = taxa4+taxa3+taxa2
    print('R$ {:.2f}'.format(total))